addons/im_livechat/models/discuss_channel.py

```diff
@@ -741,7 +749,23 @@ def _add_members(
             post_joined_message=post_joined_message,
             inviting_partner=inviting_partner,
         )
+        if not any(is_livechat_channel(channel) for channel in self):
+            return all_new_members
+        history_by_channel = defaultdict(lambda: self.env["im_livechat.channel.member.history"])
+        for history in all_new_members.livechat_member_history_ids:
+            history_by_channel[history.channel_id] |= history
         for channel in all_new_members.channel_id:
+            if histories_to_add := history_by_channel.get(channel):
+                Store(bus_channel=channel).add(
```

Comment:
It's best if the stores are built outside of the loop with lazymapping so we don't create more than one bus notif per channel.

Then I believe we no longer need 2 separate loops and everything about history can be in the first one.

Reply:
Note: probably out of scope, but it would be even better if the stores where build top-most level and passed to `_add_members` so we could add data into the same store from all the places.